weave

```diff
@@ -261,14 +261,14 @@ elif [ "$1" = "run" -a -z "$IS_LOCAL" ] ; then
     # we can't rely on our baked-in docker to support those arguments.
     shift 1
     if [ "$1" != "--without-dns" ] ; then
-        DNS_ARGS=$(exec_remote dns-args "$@" || true)
+        eval "$(exec_remote dns-args "$@" 2>/dev/null || true)"
```

Comment:
Ok, so when a user launches `./weave run -ti -h test2.weave.local ...`, he/she should get 

```
# container hostname
10.32.0.1    test2.weave.local test2
```

but if he/she launches `./weave run -ti -h test2 ...`, he/she will get

```
# container hostname
10.32.0.1    test2
```

Is that right? Shouldn't we add `test2.weave.local` too?